core/src/win32/filed/vss.cc

```diff
@@ -223,44 +210,38 @@ bool VSSClient::GetShadowPathW(const wchar_t* szFilePath,
 
 const size_t VSSClient::GetWriterCount()
 {
-  return pAlistWriterInfoText_->size();
+  return writer_info_.size();
 }
 
-const char* VSSClient::GetWriterInfo(int nIndex)
+const char* VSSClient::GetWriterInfo(size_t nIndex)
 {
-  return (char*)pAlistWriterInfoText_->get(nIndex);
+  if ( nIndex < writer_info_.size() ) {
+    return writer_info_.at(nIndex).info_text_.c_str();
+  }
+  return nullptr;
 }
 
-const int VSSClient::GetWriterState(int nIndex)
+const int VSSClient::GetWriterState(size_t nIndex)
 {
-  void* item = pAlistWriterState_->get(nIndex);
-
-  /*
-   * Eliminate compiler warnings
-   */
-#ifdef HAVE_VSS64
-  return (int64_t)(char*)item;
-#else
-  return (int)(char*)item;
-#endif
+  if ( nIndex < writer_info_.size() ) {
+    return writer_info_.at(nIndex).state_;
+  }
+  return 0;
 }
 
 void VSSClient::AppendWriterInfo(int nState, const char* pszInfo)
 {
-  pAlistWriterInfoText_->push(bstrdup(pszInfo));
-  pAlistWriterState_->push((void*)nState);
+  WriterInfo info;
```

Comment:
if WriterInfo had a ctor(int, const char*) this function could just do
``writer_info_.push_back(WriterInfo{nState, pszInfo});``